src/IO/readers.jl

```diff
@@ -0,0 +1,356 @@
+        Base.read!(reader, "meshes/$interval1"),
+        Base.read!(reader, "meshes/$interval2"),
+    )
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{EquiangularCubedSphere})
+
+Reads an object named 'name' of type 'EquiangularCubedSphere'.
+"""
+function read_new!(
+    reader::HDF5Reader,
+    name::AbstractString,
+    ::Type{EquiangularCubedSphere},
+)
+    (; file, cache) = reader
+    g = file["meshes/" * name]
+    localelementmap = eval(Meta.parse(attrs(g)["localelementmap"]))
+    ne = attrs(g)["ne"]
+    domain = attrs(g)["domain"]
+    cache[name] = Meshes.EquiangularCubedSphere(
+        Base.read!(reader, "domains/" * domain),
+        ne,
+        localelementmap(),
+    )
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{IntervalTopology})
+
+Reads an object named 'name' of type 'IntervalTopology'.
+"""
+function read_new!(
+    reader::HDF5Reader,
+    name::AbstractString,
+    ::Type{IntervalTopology},
+)
+    (; file, cache) = reader
+    g = file["topologies/" * name]
+    mesh = attrs(g)["mesh"]
+    cache[name] =
+        Topologies.IntervalTopology(Base.read!(reader, "meshes/" * mesh))
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{Topology2D})
+
+Reads an object named 'name' of type 'Topology2D'.
+"""
+function read_new!(reader::HDF5Reader, name::AbstractString, ::Type{Topology2D})
+    (; file, cache) = reader
+    g = file["topologies/" * name]
+    elemorder_matrix = HDF5.read(g, "elemorder")
+    elemorder = matrix_to_cartesianindices(elemorder_matrix)
+    mesh = attrs(g)["mesh"]
+    cache[name] =
+        Topologies.Topology2D(Base.read!(reader, "meshes/" * mesh), elemorder)
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{SpectralElementSpace1D})
+
+Reads an object named 'name' of type 'SpectralElementSpace1D'.
+"""
+function read_new!(
+    reader::HDF5Reader,
+    name::AbstractString,
+    ::Type{SpectralElementSpace1D},
+)
+    (; file, cache) = reader
+    g = file["spaces/" * name]
+    quadrature_num_points = attrs(g)["quadrature_num_points"]
+    quadrature_style = eval(
+        Meta.parse(
+            attrs(g)["quadrature_type"] * "{$(quadrature_num_points)}()",
+        ),
+    )
+    topology = attrs(g)["topology"]
+    cache[name] = Spaces.SpectralElementSpace1D(
+        Base.read!(reader, "topologies/" * topology),
+        quadrature_style,
+    )
+    return nothing
+end
+
+"""
+    read_new!(reader, name, ::Type{SpectralElementSpace2D})
+
+Reads an object named 'name' of type 'SpectralElementSpace2D'.
+"""
+function read_new!(
+    reader::HDF5Reader,
+    name::AbstractString,
+    ::Type{SpectralElementSpace2D},
+)
+    (; file, cache) = reader
+    g = file["spaces/" * name]
+    quadrature_num_points = attrs(g)["quadrature_num_points"]
+    quadrature_style = eval(
+        Meta.parse(
+            attrs(g)["quadrature_type"] * "{$(quadrature_num_points)}()",
+        ),
+    )
```

Comment:
branch instead of `eval` (can write a helper function to avoid duplicating code with above)